Sources/Authgear.swift

```diff
@@ -92,6 +92,10 @@ public protocol AuthgearDelegate: AnyObject {
     func authgearSessionStateDidChange(_ container: Authgear, reason: SessionStateChangeReason)
 }
 
+public extension AuthgearDelegate {
+    func authgearSessionStateDidChange(_ container: Authgear, reason: SessionStateChangeReason) {}
```

Comment:
`optional` is objective-c keyword, I tried to add `@objc` to the protocol like this
```
@objc public protocol AuthgearDelegate: AnyObject {
    @objc optional func authgearSessionStateDidChange(_ container: Authgear, reason: SessionStateChangeReason)
    @objc optional func sendWeChatAuthRequest(_ state: String)
}
```
But turns out that we cannot mark function `authgearSessionStateDidChange` with `@objc`, because `SessionStateChangeReason` cannot be represented in objective-c. Here is the error:

```
Method cannot be marked @objc because the type of the parameter 2 cannot be represented in Objective-C
```

So I chose to use a pure swift way to do that,
https://stackoverflow.com/questions/30979219/optional-can-only-be-applied-to-members-of-an-objc-protocol